processing/src/main/java/io/druid/segment/incremental/IncrementalIndexStorageAdapter.java

```diff
@@ -576,7 +583,7 @@ public ValueMatcher makeValueMatcher(String dimension, String value)
       }
       final IncrementalIndex.DimDim dimDim = index.getDimension(dimension);
       if (!dimDim.contains(value)) {
-        if (value == null || "".equals(value)) {
+        if (Strings.isNullOrEmpty(value)) {
```

Comment:
if DimDim is already doing the conversion from null to empty, is there a reason to keep checking for both?